src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialForm.tsx

```diff
@@ -119,8 +131,9 @@ const MaterialForm: React.FC<MaterialFormProps> = ({ submitText, onSubmit, defau
 
   const onSubmitWrapper = (data: MaterialFormInput): void => {
     const price = Math.round(data.price * 100);
-    const subtotal = data.unitName ? price : data.quantity * price;
-    onSubmit({ ...data, subtotal: subtotal, price: price });
+    const quantity = Number(data.quantity);
+    const subtotal = data.unitName ? price : quantity * price;
```

Comment:
I think I added this in, but I can't see any reason why it should be here. subtotal should just be quantity * price